### PR TITLE
Add endpoint for getting a file by ID

### PR DESCRIPTION
**Pull request by Alex-Radu**

### CODE REVIEW

**Comment by achingbrain** [comment]
Looks good.  Could you add some tests please?

**Comment by Alex-Radu** [comment]
I didn't want to add tests intentionally because I don't think the provide much/any value. I am talking about all/most of the tests inside the `api` files. 

I didn't look over every single tests, but I've seen that most of them test that if we call `requestFun`, for example, with a certain object, the values inside that object are actually used, which we're already testing here: [requestFun Test](https://github.com/Mendeley/mendeley-javascript-sdk/blob/master/test/spec/utilities/utilities.spec.js#L40)

Also, some of the tests are outdated, for example [this](https://github.com/Mendeley/mendeley-javascript-sdk/blob/master/test/spec/api/files.spec.js#L119) will be true no matter what the value of the accept header is because of [this](https://github.com/Mendeley/mendeley-javascript-sdk/blob/master/lib/request.js#L48). I am guessing that somewhere along the way, the Accept header got a default value and these tests were not updated.

In any case, I will add the tests for consistency's sake, but I would also remove most of them. 😄

**Comment by customcommander** [comment]
I would tend to agree with @Alex-Radu.

Those unit tests are simply testing `requestFun` in disguise over and over again. Considering that @charliehw has introduced proper tests for it in https://github.com/Mendeley/mendeley-javascript-sdk/commit/13c5d80440409320116fb28dbf60703449f9f6aa, it's probably okay.

**Comment by achingbrain** [comment]
It is a fairly convincing argument.

File: lib/api/files.js [hunk 1]
```diff
@@ -47,10 +47,23 @@ module.exports = function files(options) {
             baseUrl: options.baseUrl,
             method: 'GET',
             resource: '/files?document_id={id}',
-            args: ['id'],
-            headers: {
-              'Accept': MIME_TYPES.FILE
```
Comment (charliehw) [inline lib/api/files.js]: Why remove the Accept header from GET /files ?

Comment (Alex-Radu) [inline lib/api/files.js]: Good spot - did that when checking something with the tests. It's back now

File: lib/api/files.js [hunk 2]
```diff
@@ -54,6 +54,22 @@ module.exports = function files(options) {
...
 
         /**
+         * Get a file
+         *
+         * @method
+         * @memberof api.files
+         * @param {string} id - A file UUID
+         * @returns {promise}
+         */
+        retrieve: utils.requestFun({
```
Comment (achingbrain) [inline lib/api/files.js]: Why does this not have an `Accept` header?

Comment (Alex-Radu) [inline lib/api/files.js]: I've looked at the swagger docs and they don't specify any required header

Comment (achingbrain) [inline lib/api/files.js]: Ah, true.  I was looking at the wrong resource.

**Pull request merged**